lib/mongoid/clients/options.rb

```diff
@@ -44,12 +44,12 @@ def persistence_options
 
       def mongo_client
         tmp = persistence_options
-        if opts = tmp && tmp.dup
+        if (opts = tmp && !tmp.empty? && tmp.dup)
```

Comment:
The bug was found because Mongoid code gave empty hashes as `persistence_options`.

The new code handles those empty hashes as if there are no persistence options.